### ISSUE TITLE
[CI] RemoteClusterClientTests testEnsureWeReconnect failing with NoSuchRemoteClusterException

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#52029]
This test has failed 4 times in the past three days after passing basically 100% of the time for the past month. Look suspicious. Happening on both `master` and `7.x`.

```
:server:test » org.elasticsearch.transport.RemoteClusterClientTests » testEnsureWeReconnect (1.662s)
org.elasticsearch.transport.NoSuchRemoteClusterException: no such remote cluster: [test]
java.util.concurrent.ExecutionException: org.elasticsearch.transport.NoSuchRemoteClusterException: no such remote cluster: [test]Open stacktrace
Caused by: org.elasticsearch.transport.NoSuchRemoteClusterException: no such remote cluster: [test]Open stacktrace
[2020-02-07T11:10:26,252][INFO ][o.e.t.RemoteClusterClientTests] [testEnsureWeReconnect] before test
[2020-02-07T11:10:26,808][INFO ][o.e.t.TransportService   ] [testEnsureWeReconnect] publish_address {127.0.0.1:14000}, bound_addresses {[::1]:14000}, {127.0.0.1:14000}
[2020-02-07T11:10:27,048][INFO ][o.e.t.TransportService   ] [testEnsureWeReconnect] publish_address {127.0.0.1:14001}, bound_addresses {[::1]:14001}, {127.0.0.1:14001}
[2020-02-07T11:10:27,519][INFO ][o.e.t.RemoteClusterClientTests] [testEnsureWeReconnect] after test
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.transport.RemoteClusterClientTests.testEnsureWeReconnect" -Dtests.seed=EE4961D2D47604F4 -Dtests.security.manager=true -Dtests.locale=zh-Hans-SG -Dtests.timezone=Asia/Pontianak -Dcompiler.java=13
```

This same time period has also seen a rather significant jump in average test execution times so perhaps there is something going on here.
![image](https://user-images.githubusercontent.com/4106672/74005594-3da88900-492e-11ea-911d-e35cdc2c28c7.png)

https://gradle-enterprise.elastic.co/scans/tests?failures.failureClassification=non_verification&list.offset=0&list.size=50&list.sortColumn=startTime&list.sortOrder=desc&search.buildToolType=gradle&search.buildToolType=maven&search.startTimeMax=1581055741639&search.startTimeMin=1580450941631&search.tags=CI&search.tags=not:nested&search.tags=not:pull-request&tests.container=org.elasticsearch.transport.RemoteClusterClientTests&tests.sortField=FAILED&tests.test=testEnsureWeReconnect&tests.unstableOnly&trends.section=overview&trends.timeResolution=day&viewer.tzOffset=-480

**Comment by albertzaharovits** [comment]
two more such failures on master
https://gradle-enterprise.elastic.co/s/gjthqdvfktvvg
https://gradle-enterprise.elastic.co/s/bp7nr62mxmski
Going to mute

**Comment by albertzaharovits** [comment]
Only muted in master d4c609b

**Comment by pugnascotia** [comment]
Failed on 7.6 too - https://gradle-enterprise.elastic.co/s/p6yxu3p22ni52

**Comment by mayya-sharipova** [comment]
Another failure on master today. Reopening the issue.

Log: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob+fast+part1/4097/console
Build Scans: https://gradle-enterprise.elastic.co/s/hijlawbrzpxnu

Stack trace:
```
09:23:34 org.elasticsearch.transport.RemoteClusterClientTests > testEnsureWeReconnect FAILED
09:23:34     java.util.concurrent.ExecutionException: org.elasticsearch.transport.NoSuchRemoteClusterException: no such remote cluster: [test]
09:23:34         at __randomizedtesting.SeedInfo.seed([C01B5CF26B407FDA:1498B3C6DC018873]:0)
09:23:34         at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:266)
09:23:34         at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:253)
09:23:34         at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:87)
09:23:34         at org.elasticsearch.transport.RemoteClusterClientTests.testEnsureWeReconnect(RemoteClusterClientTests.java:111)
```

**Comment by tbrooks8** [comment]
I have merged https://github.com/elastic/elasticsearch/pull/54934. This PR may not completely fix this issue. But it will help expose the underlying cause if the test fails again. At the moment we are in wait and see mode to see if this test continues to fail.

**Comment by tlrx** [comment]
@tbrooks8 The test failed today on CI but with the exception you added in  #54934: 

```
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Unable to open any connections to remote cluster [test]
        at __randomizedtesting.SeedInfo.seed([7E33386F2D392BD:D360DCB245926514]:0)
        at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:266)
        at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:253)
        at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:87)
        at org.elasticsearch.transport.RemoteClusterClientTests.testEnsureWeReconnect(RemoteClusterClientTests.java:113)

        Caused by:
        java.lang.IllegalStateException: Unable to open any connections to remote cluster [test]
            at org.elasticsearch.transport.SniffConnectionStrategy$SniffClusterStateResponseHandler.handleNodes(SniffConnectionStrategy.java:367)
            at org.elasticsearch.transport.SniffConnectionStrategy$SniffClusterStateResponseHandler.handleResponse(SniffConnectionStrategy.java:329)
            at org.elasticsearch.transport.SniffConnectionStrategy$SniffClusterStateResponseHandler.handleResponse(SniffConnectionStrategy.java:309)
            at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1101)
            at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1101)
            at org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:206)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:691)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            at java.base/java.lang.Thread.run(Thread.java:834)
```

I can't reproduce the issue locally.
Build scan: https://gradle-enterprise.elastic.co/s/nni5qkxjegck6

**Comment by iverase** [comment]
Two more failures in master:

https://gradle-enterprise.elastic.co/s/iycqzcwkouene
https://gradle-enterprise.elastic.co/s/iycqzcwkouene

**Comment by markharwood** [comment]
A case on 7.7 this morning
https://gradle-enterprise.elastic.co/s/5rxifwbiqaudq

**Comment by tbrooks8** [comment]
I believe this has been fixed by #56654. Closing.
